src/libs/HttpUtils.ts

```diff
@@ -172,6 +174,25 @@ function xhr(command: string, data: Record<string, unknown>, type: RequestType =
     return processHTTPRequest(url, type, formData, abortSignalController?.signal);
 }
 
+/**
+ * Ensures no value of type `object` other than Blob or its subclasses is passed to XMLHttpRequest.
+ * Otherwise, it will be incorrectly serialized as `[object Object]` and cause an error on Android.
+ * See https://github.com/Expensify/App/issues/45086
+ */
+function validateFormDataParameter(command: string, key: string, value: unknown) {
+    // eslint-disable-next-line @typescript-eslint/no-shadow
+    const isValid = (value: unknown) => value === null || typeof value !== 'object' || value instanceof Blob;
+    if (Array.isArray(value)) {
+        if (value.every(isValid)) {
+            return;
+        }
+    } else if (isValid(value)) {
+        return;
+    }
```

Comment:
```suggestion
    if (isValid(value)) {
        return;
    }
```
I think simply `isValid(value)` should be fine here, otherwise it can be a nested-level array too.

Reply:
Added support for nested arrays.

Reply:
Sorry but I was thinking of only do the validation for TopLevel, can you plz revert to the previous one where you applied it to one level bottom? Bcoz instead of making this complex that seems to be the structure for mostly all requests.

Reply:
I think it is fine as it is now, as it covers all possible cases. Note that the `isTopLevel` param wasn't added specifically for nested arrays. Files are not allowed within arrays, so even if we make it one level deep, it will be pretty much the same code just a bit reordered

```ts
function validateFormDataParameter(command: string, key: string, value: unknown) {
    // eslint-disable-next-line @typescript-eslint/no-shadow
    const isValid = (value: unknown) => value === null || typeof value !== 'object';
    // Native platforms only require the value to include the `uri` property.
    // Optionally, it can also have a `name` and `type` props.
    // On other platforms, the value must be an instance of `Blob`.
    // eslint-disable-next-line @typescript-eslint/no-shadow
    const isValidTopLevel = (value: unknown) => isValid(value) || isNativePlatform ? 'uri' in value && !!value.uri : value instanceof Blob;

    if (Array.isArray(value)) {
        if (value.every(isValid)) {
            return;
        }
    } else if (isValidTopLevel(value)) {
        return;
    }
    // eslint-disable-next-line no-console
    console.warn(`An unsupported value was passed to command '${command}' (parameter: '${key}'). Only Blob and primitive types are allowed.`);
}
```